server/src/test/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioningTests.java

```diff
@@ -1042,6 +1045,31 @@ void updateLastLogonTime() {
         assertNotNull(user.getLastLogonTime());
     }
 
+    @Test
+    void cannotCreateMaxUserLimit() {
+        IdentityZoneHolder.get().getConfig().getUserConfig().setMaxUsers(10);
+        ScimUser scimUser = new ScimUser("user-id-1", "user1@example.com", "User", "Example");
+        ScimUser.Email email = new ScimUser.Email();
+        email.setValue("user@example.com");
+        scimUser.setEmails(Collections.singletonList(email));
+        scimUser.setPassword("password");
+        scimUser.setOrigin(OriginKeys.UAA);
+        for (int i = 2; i < 12; i++) {
+            scimUser = jdbcScimUserProvisioning.create(scimUser, IdentityZoneHolder.getCurrentZoneId());
+            scimUser.setId("user-id-" + 1);
+            scimUser.setUserName("user" +i+ "@example.com");
+            scimUser.setPassword("password");
+        }
+        scimUser.setPassword("password");
+        ScimUser finalScimUser = scimUser;
+        assertThrowsWithMessageThat(
+            InvalidScimResourceException.class,
+            () -> jdbcScimUserProvisioning.create(finalScimUser, IdentityZoneHolder.getCurrentZoneId()),
+            containsString("The maximum allowed numbers of users: 10 is reached already in Identity Zone")
+        );
+        IdentityZoneHolder.get().getConfig().getUserConfig().setMaxUsers(-1);
```

Comment:
[nitpick] Should we move this to an `AfterEach` handler?

Reply:
Otherwise, the value would not be reset if an exception is thrown in one of the previous statements